### PR TITLE
Implement LWG-3949: `atomic<bool>`'s trivial destructor dropped in C++17 spec wording

### LINKED ISSUE #4164: LWG-3949 `atomic<bool>`'s trivial destructor dropped in C++17 spec wording
LWG-3949 `atomic<bool>`'s trivial destructor dropped in C++17 spec wording

This is already implemented and tested, we just need to drop "(and bool)" from this comment:

https://github.com/microsoft/STL/blob/46843b3bf320c6070ff63e978936bcf898e5a2bc/tests/std/tests/Dev11_0863628_atomic_compare_exchange/test.cpp#L326-L335

### PR DESCRIPTION
**Pull request by fsb4000**
Fixes #4164

### CODE REVIEW

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks! This comment change needs only one maintainer approval.

I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
:atom_symbol: :tada: :white_check_mark:
===

**Pull request merged**
